proxy-support/README.md

```diff
@@ -0,0 +1,273 @@
+  DNS:
+    IP: 10.96.0.10
+proxy:
+  noProxy: cluster.local,100.64.0.0/13,100.96.0.0/11,178.170.32.59,172.16.0.0/16,internal.corpdomain.net,example.com,svc,127.0.0.1,localhost
+```
+
+[`cluster-apps-operator`] also creates a `secret` which contains some infrastructure specific values ([for `openstack` and `vcd`](https://github.com/giantswarm/cluster-apps-operator/blob/master/service/controller/resource/clustersecret/desired.go#L42-L65)).
+The `secret` for each WC is called `<clusterName>-cluster-values`.
+With the extended proxy implementation, the secret now contains a new `cluster` section with a `proxy` sub section:
+
+```yaml
+cluster:
+  proxy:
+    httpProxy: http://myProxyUser:mySuperSecretPassword@192.168.52.220:3128
+    httpsProxy: http://myProxyUser:mySuperSecretPassword@192.168.52.220:3128
+```
+
+This `secret` and `configmap` can now be used to pass these information into `Apps` which get applied into a WC.
+As `cert-manager` needs this data, the manifest should now look like:
+
+```yaml
+apiVersion: application.giantswarm.io/v1alpha1
+kind: App
+[...]
+spec:
+  catalog: default-test
+  config:
+    configMap:
+      name: mycluster-cluster-values
+      namespace: org-giantswarm
+    secret:
+      name: mycluster-cluster-values
+      namespace: org-giantswarm
+[...]
+```
+
+As we have these information on a cluster-base, we are now able to create a new `secret`, called `<clusterName>-systemd-proxy` which is a rendered version of [this](#containerd) systemd drop-in. In our example, the `secret` now looks like
+
+```yaml
+[Service]
+Environment="HTTP_PROXY=http://myProxyUser:mySuperSecretPassword@192.168.52.220:3128"
+Environment="http_proxy=http://myProxyUser:mySuperSecretPassword@192.168.52.220:3128"
+Environment="HTTPS_PROXY=http://myProxyUser:mySuperSecretPassword@192.168.52.220:3128"
+Environment="https_proxy=http://myProxyUser:mySuperSecretPassword@192.168.52.220:3128"
+Environment="NO_PROXY=172.16.0.0/16,internal.corpdomain.net,example.com"
+Environment="no_proxy=172.16.0.0/16,internal.corpdomain.net,example.com"
+```
+
+This secret can now be referenced in a `kubeadmconfig`, e.g.:
+
+```yaml
+files:
+  - path: /etc/systemd/system/containerd.service.d/http-proxy.conf
+    permissions: "0600"
+    contentFrom:
+      secret:
+        name: <clusterName>-systemd-proxy
+        key: containerdProxy   
+```
+
+By doing this, `ClusterAPI` is waiting for the `secret` to get created. Once the `secret` got created, it will be used by creating the machines (and injecting this data via `cloud-init` or `ignition`).
+
+As `cluster-apps-operator` takes care of creating `deployment/chart-operator` in the WC with the values from `(secret|configmap)/<clusterName>-cluster-values` as input, `chart-operator` in every WC automatically get a valid proxy configuration.
+
+## To be discussed
+
+- Customers are able to see the proxy configuration (especially the secret part) within a `WC`. Is this a security issue?
```

Comment:
@giantswarm/team-shield i'm interested in your opinion about this

Reply:
the weird thing to me is that the proxy does not seem to differentiate between proxy admins (people who should see config) and proxy users (people who just need to use the proxy). I guess login they want so that no one else can use the proxy, who should maybe not have external access? otherwise it would feel wrong

Reply:
Shield discussed this ([thread](https://gigantic.slack.com/archives/C02FL4EAADD/p1668625800984809)).

In general, we think it's necessary to expose proxy information to workloads which need it, so it has to work this way right now. We also aren't limited by this approach in the future, since using Kyverno policies it would also be possible to refine which workloads should be mutated to include the proxy information (if, for example, some pods are not trusted with that information, or if different proxy information should be injected into different pods).

As Puja mentioned, there is one caveat: I understood that any credentials contained in the proxy variables were required for a workload to use a proxy (i.e. the proxy requires authentication). However, Puja understood it differently, and if that is NOT the case, and the credentials are somehow "admin" credentials for actually administering the proxy, then we should probably skip straight to splitting the policies and only injecting those specific credentials where needed.

Reply:
The current implementation doesn't fullfil this need, if the proxy users shouldn't see any kind of proxy configuration. 

If this is something which should be abstracted for the users and they shouldn't see this configuration we have to think about some `transparent proxy` solutions (by implementing some iptables rules + a installation specific proxy server which act as "man in the middle" with the only purpose of having corporate proxy credentials stored).

Reply:
> The current implementation doesn't fullfil this need, if the proxy users shouldn't see any kind of proxy configuration.

Right, I meant if a workload did not need to use the proxy and should not see the proxy info, not that it needed to use the proxy but was not trusted. I agree that would be a separate problem